trunk/src/app/srs_app_http_conn.hpp

```diff
@@ -136,8 +136,18 @@ class SrsHttpxConn : public ISrsConnection, public ISrsStartable, public ISrsHtt
     SrsHttpConn* conn;
     // We should never enable the stat, unless HTTP stream connection requires.
     bool enable_stat_;
-public:
-    SrsHttpxConn(bool https, ISrsResourceManager* cm, ISrsProtocolReadWriter* io, ISrsHttpServeMux* m, std::string cip, int port);
+    // ssl key & cert file
+    const std::string ssl_key_file;
+    const std::string ssl_cert_file;
+    
+public:
+    SrsHttpxConn(ISrsResourceManager* cm,
+                 ISrsProtocolReadWriter* io,
+                 ISrsHttpServeMux* m,
+                 std::string cip,
+                 int port,
+                 const std::string& ssl_key_file = "",
+                 const std::string& ssl_cert_file = "");
```

Comment:
Multiple lines is very difficult to read, no need to use `const std::string&`, just use `string`, because it's not hot path and not bottleneck. We must keep it easy to read:

```cpp

SrsHttpxConn(ISrsResourceManager* cm, ISrsProtocolReadWriter* io, ISrsHttpServeMux* m, std::string cip, int port, string key, string cert);
```

No default value generally.

Reply:
fixed.